core/src/main/java/fr/sncf/osrd/envelope_sim/pipelines/MaxSpeedEnvelope.java

```diff
@@ -28,7 +28,12 @@ static boolean increase(double prevPos, double prevSpeed, double nextPos, double
     private static Envelope addBrakingCurves(EnvelopeSimContext context, Envelope mrsp) {
         var builder = OverlayEnvelopeBuilder.backward(mrsp);
         var cursor = EnvelopeCursor.backward(mrsp);
+        var lastPosition = mrsp.getEndPos();
         while (cursor.findPartTransition(MaxSpeedEnvelope::increase)) {
+            if (cursor.getPosition() > lastPosition) {
+                cursor.nextPart();
+                continue;
+            }
```

Comment:
please add a comment: "if a breaking curve starts inside the next breaking curve, it is completely hidden"